### PR TITLE
support action input for release draft body

### PR DESCRIPTION
**Pull request by jetersen**
@jglick 

https://github.com/jenkins-infra/github-reusable-workflows/pull/2#issuecomment-1145473681

### CODE REVIEW

**Review by timja — APPROVED** [review]

File: run.sh [hunk 1]
```diff
@@ -2,7 +2,11 @@
 set -euxo pipefail
 if [ $GITHUB_EVENT_NAME = check_run ]
 then
-    RESULT=$(gh api /repos/$GITHUB_REPOSITORY/releases | jq -e -r '.[] | select(.draft == true and .name == "next") | .body' | egrep "$INTERESTING_CATEGORIES" || echo 'failed')
+    if [ -z "$RELEASE_DRAFT_BODY" ]; then
+        RESULT=$(gh api /repos/$GITHUB_REPOSITORY/releases | jq -e -r '.[] | select(.draft == true and .name == "next") | .body' | egrep "$INTERESTING_CATEGORIES" || echo 'failed')
+    else
+        RESULT=$(echo "$RELEASE_DRAFT_BODY" | egrep "$INTERESTING_CATEGORIES" || echo 'failed')
+    fi
```
Comment (jglick) [inline run.sh:9]:
```suggestion
    if [ -z "$RELEASE_DRAFT_BODY" ]
    then
        RELEASE_DRAFT_BODY="$(gh api /repos/$GITHUB_REPOSITORY/releases | jq -e -r '.[] | select(.draft == true and .name == "next") | .body')"
    fi
    RESULT=$(echo "$RELEASE_DRAFT_BODY" | egrep "$INTERESTING_CATEGORIES" || echo 'failed')
```

**Review by jglick — APPROVED** [review]

File: run.sh (hunk 1 above)
Comment (jetersen) [reply run.sh:9]:
```suggestion
    if [ -z "$RELEASE_DRAFT_BODY" ]; then
        RELEASE_DRAFT_BODY="$(gh api /repos/$GITHUB_REPOSITORY/releases | jq -e -r '.[] | select(.draft == true and .name == "next") | .body')"
    fi
	    
    RESULT=$(echo "$RELEASE_DRAFT_BODY" | egrep "$INTERESTING_CATEGORIES" || echo 'failed')
```

**Review by jglick — APPROVED** [review]
Looks right. Tested yet in context of the workflow?

**Comment by jglick** [comment]
TBD whether this has any effect on https://github.com/jenkins-infra/jenkins-maven-cd-action/issues/4.

File: action.yaml [hunk 2]
```diff
@@ -11,13 +11,18 @@ inputs:
       By default excludes 📦📝👻🚦 under the assumption these do not normally merit a release.
       An output of interesting = true/false will be set, the 'workflow_dispatch' (explicit release) is always 'interesting'.
     default: '[💥🚨🎉🐛⚠🚀🌐👷]|:(boom|tada|construction_worker):'
+  RELEASE_DRAFT_BODY:
+    required: false
+    description: |
+      Release draft body produced by release drafter github action
```
Comment (jglick) [inline action.yaml:17]:
```suggestion
      Release draft body produced by `release-drafter/release-drafter` action; defaults to looking for a draft release named `next`
```

**Review by jglick — COMMENTED** [review]
https://github.com/jenkinsci/jenkins-infra-test-plugin/runs/6838449291?check_suite_focus=true#step:4:104 the variable is not being bound

**Review by jglick — CHANGES_REQUESTED** [review]
currently broken

**Comment by jetersen** [comment]
@jglick fixed:
https://github.com/jenkinsci/jenkins-infra-test-plugin/runs/6840951645?check_suite_focus=true

File: action.yaml [hunk 3]
```diff
@@ -11,13 +11,20 @@ inputs:
...
     value: ${{ steps.interesting-category.outputs.interesting }}
 runs:
   using: composite
-  steps: 
+  steps:
     - run: GITHUB_TOKEN=${{ inputs.GITHUB_TOKEN }} INTERESTING_CATEGORIES='${{ inputs.INTERESTING_CATEGORIES }}' $GITHUB_ACTION_PATH/run.sh
       id: interesting-category
       shell: bash
+      env:
+        RELEASE_DRAFT_BODY: ${{ inputs.RELEASE_DRAFT_BODY }}
```
Comment (jetersen) [inline action.yaml:30]: Perhaps move GITHUB_TOKEN and INTERESTING_CATEGORIES to the ENV block?

Comment (jglick) [reply action.yaml:30]: Sure, I think I just did not realize that option existed when I created the original.

File: run.sh [hunk 4]
```diff
@@ -2,7 +2,11 @@
 set -euxo pipefail
 if [ $GITHUB_EVENT_NAME = check_run ]
 then
-    RESULT=$(gh api /repos/$GITHUB_REPOSITORY/releases | jq -e -r '.[] | select(.draft == true and .name == "next") | .body' | egrep "$INTERESTING_CATEGORIES" || echo 'failed')
+    if [ -z "$RELEASE_DRAFT_BODY" ]; then
+        RELEASE_DRAFT_BODY="$(gh api /repos/$GITHUB_REPOSITORY/releases | jq -e -r '.[] | select(.draft == true and .name == "next") | .body')"
+    fi
+
+    RESULT=$(echo "$RELEASE_DRAFT_BODY" | egrep "$INTERESTING_CATEGORIES" || echo 'failed')
     if [[ $RESULT != 'failed' ]]; then
```
Comment (jglick) [inline run.sh:10]: BTW I think this could be simplified:
```suggestion
    if echo "$RELEASE_DRAFT_BODY" | egrep "$INTERESTING_CATEGORIES"
    then
```

**Review by jglick — APPROVED** [review]

**Comment by jetersen** [comment]
@jglick Resolved your suggestions :)

**Comment by jetersen** [comment]
@jglick I don't know who is maintainer on this and can merge it 😅 

Perhaps @timja or one of the other infra people @lemeurherve

**Comment by jglick** [comment]
I have write access. Let me retest it.

**Comment by lemeurherve** [comment]
> I don't know who is maintainer on this and can merge it 😅

@jetersen would you like to be added as maintainer on this repository?

Great work on the github actions by the way!

**Comment by jetersen** [comment]
@lemeurherve sure I believe @timja already added me to the other action repos 😄

**Comment by lemeurherve** [comment]
Added, you should receive an invitation shortly.

**Pull request merged**